fix: Add `DownloadContent` to support downloads of any size

sheeeng opened the pull request:
Fix https://github.com/google/go-github/issues/3810.
Fix https://github.com/google/go-github/issues/2480.

sheeeng committed:
fix: download contents

Fix https://github.com/google/go-github/issues/3810.
Fix https://github.com/google/go-github/issues/2480.

Signed-off-by: Leonard Sheng Sheng Lee <leonard.sheng.sheng.lee@gmail.com>

gmlewis reviewed (commented):
My biggest concern is that we now have:
* DownloadContent
* DownloadContents
* DownloadContentsWithMeta
* GetContent
* GetContents

How is a user supposed to understand which one to choose?

sheeeng committed:
fix: download contents

Fix https://github.com/google/go-github/issues/3810.
Fix https://github.com/google/go-github/issues/2480.

Signed-off-by: Leonard Sheng Sheng Lee <leonard.sheng.sheng.lee@gmail.com>

sheeeng commented:
> * DownloadContents
> * DownloadContentsWithMeta

These two are existing functions. Do we want to expose `DownloadContents` only for use?

sheeeng committed:
feat: prioritize downloadcontents

Delete DownloadContentsWithMeta function and its references.

Signed-off-by: Leonard Sheng Sheng Lee <leonard.sheng.sheng.lee@gmail.com>

sheeeng commented:
Setting this PR as draft as waiting for more feedback.

gmlewis commented:
Closing, as this was resolved by #4153.
